--- a/lib/AppInfo/Application.php
+++ b/lib/AppInfo/Application.php
@@ -14,6 +14,7 @@
 use OCP\AppFramework\Bootstrap\IRegistrationContext;
 use OCP\AppFramework\Bootstrap\IBootContext;
 use OCP\IL10N;
+use OCP\INavigationManager;
 use OCP\IUser;
 use OCP\User\Events\PostLoginEvent;
 use OCP\User\Events\BeforeUserLoggedOutEvent;
@@ -37,7 +38,8 @@
 			'PageController', function($c) {
 				return new PageController(
 					$c->query('AppName'),
-					$c->query('Request')
+					$c->query('Request'),
+					$c->query(INavigationManager::class)
 				);
 			}
 		);
--- a/lib/Controller/PageController.php
+++ b/lib/Controller/PageController.php
@@ -7,9 +7,19 @@
 
 use OCP\AppFramework\Controller;
 use OCP\AppFramework\Http\TemplateResponse;
+use OCP\INavigationManager;
+use OCP\IRequest;
 
 class PageController extends Controller
 {
+	public function __construct(
+		string $appName,
+		IRequest $request,
+		private INavigationManager $navigationManager,
+	) {
+		parent::__construct($appName, $request);
+	}
+
 	/**
 	 * @NoAdminRequired
 	 * @NoCSRFRequired
@@ -28,7 +38,7 @@
 		}
 
 		if (!$bAdmin && $config->getAppValue('snappymail', 'snappymail-no-embed')) {
-			\OC::$server->getNavigationManager()->setActiveEntry('snappymail');
+			$this->navigationManager->setActiveEntry('snappymail');
 			\OCP\Util::addScript('snappymail', 'snappymail');
 			\OCP\Util::addStyle('snappymail', 'style');
 			SnappyMailHelper::startApp();
@@ -43,7 +53,7 @@
 			return $response;
 		}
 
-		\OC::$server->getNavigationManager()->setActiveEntry('snappymail');
+		$this->navigationManager->setActiveEntry('snappymail');
 
 		\OCP\Util::addStyle('snappymail', 'embed');
 
