--- a/lib/AppInfo/Application.php
+++ b/lib/AppInfo/Application.php
@@ -80,26 +80,27 @@
 
 		$dispatcher = $context->getAppContainer()->query('OCP\EventDispatcher\IEventDispatcher');
 		$dispatcher->addListener(PostLoginEvent::class, function (PostLoginEvent $Event) {
-/*
 			$config = \OC::$server->getConfig();
-			// Only store the user's password in the current session if they have
-			// enabled auto-login using Nextcloud username or email address.
-			if ($config->getAppValue('snappymail', 'snappymail-autologin', false)
-			 || $config->getAppValue('snappymail', 'snappymail-autologin-with-email', false)) {
-*/
-				$sUID = $Event->getUser()->getUID();
-				\OC::$server->getSession()['snappymail-nc-uid'] = $sUID;
-				\OC::$server->getSession()['snappymail-passphrase'] = SnappyMailHelper::encodePassword($Event->getPassword(), $sUID);
-/*
+			if ($Event->isTokenLogin()
+			 || (!$config->getAppValue('snappymail', 'snappymail-autologin', false)
+			  && !$config->getAppValue('snappymail', 'snappymail-autologin-with-email', false))) {
+				return;
 			}
-*/
+
+			$sUID = $Event->getUser()->getUID();
+			$session = \OC::$server->getSession();
+			$session->set('snappymail-nc-uid', $sUID);
+			$session->set(
+				'snappymail-passphrase',
+				SnappyMailHelper::encodePassword($Event->getPassword(), $sUID)
+			);
 		});
 
 		$dispatcher->addListener(BeforeUserLoggedOutEvent::class, function (BeforeUserLoggedOutEvent $Event) {
-			// https://github.com/nextcloud/server/issues/36083#issuecomment-1387370634
-//			\OC::$server->getSession()['snappymail-passphrase'] = '';
+			$session = \OC::$server->getSession();
+			$session->remove('snappymail-passphrase');
+			$session->remove('snappymail-nc-uid');
 			SnappyMailHelper::loadApp();
-//			\RainLoop\Api::Actions()->Logout(true);
 			\RainLoop\Api::Actions()->DoLogout();
 		});
 
